### PR TITLE
Optimize

### PR DESCRIPTION
**Pull request by joshuabradley012**
## Trello Ticket
https://trello.com/c/ObKgNM99/26-josh-optimize-couchdb-and-postgres

## How to Setup
```
npm i
```

## Purpose of Pull Request
Optimize couchdb and postgres with indexes to make queries execute in less than 50ms.

### CODE REVIEW

File: server/db/generate-record.js [hunk 1]
```diff
@@ -23,32 +23,34 @@ const randomIntBetween = (min, max) => Math.floor(randomFloatBetween(min, max));
...
-generate.transit = ({ id }) => ({
+generate.transit = (id, locationId) => ({
   _id: id,
-  name: randomWordBetween(4, 10),
+  recordType: 'transitOption',
+  locationId: locationId,
+  name: randomWordBetween(1, 3),
   type: randomTransitType(),
-  averageCost: randomCost(),
+  averageCost: randomIntBetween(5, 100),
```
Comment (manedurphy) [inline server/db/generate-record.js:53]: nicely structured document objects

File: server/db/seed-couch.js [hunk 2]
```diff
@@ -62,10 +60,7 @@ const createJsonRecords = async () => {
 const createSeedScript = async () => {
   await fs.writeFile(shellScriptFile, `#!/bin/bash
     for filename in ${recordsDir}/*.json; do
-      curl --silent -d @$filename
-        -H "Content-type: application/json"
-        -X POST '${couchDbConnectionString}/${process.env.CDDATABASE}/_bulk_docs?batch=ok'
-        >/dev/null
+      curl --silent -d @$filename -H "Content-Type: application/json" -X POST '${couchDbConnectionString}/${process.env.CDDATABASE}/_bulk_docs?batch=ok' >/dev/null
```
Comment (manedurphy) [inline server/db/seed-couch.js:63]: love the bash script

File: server/db/seed-postgres.js [hunk 3]
```diff
@@ -87,12 +87,12 @@ const insertRecords = async client => {
...
     for (let i = 0; i < RECORDS_SQRT && locationId < RECORDS; i++) {
-      const location = generate.location({ id: locationId });
+      const location = generate.location();
       query += `EXECUTE insert_location(${locationId}, '${location.name}', '${location.address}', '${location.coordinates}');`;
 
       for (let j = 0; j < generate.randomIntBetween(5, 10); j++) {
-        const transit = generate.transit({ id: transitOptionId });
-        query += `EXECUTE insert_transit_option(${transitOptionId}, ${locationId}, '${transit.name}', '${transit.type}', '${transit.averageCost}');`;
+        const transit = generate.transit({ id: transitOptionId, locationId: location._id });
+        query += `EXECUTE insert_transit_option(${transitOptionId}, ${locationId}, '${transit.name}', '${transit.type}', ${transit.averageCost});`;
```
Comment (manedurphy) [inline server/db/seed-postgres.js:95]: Nice use of raw queries. How long did it take to seed 10 M records?

**Review by manedurphy — APPROVED** [review]
As usual, very clean and understandable code. Had a question about your Postgres seed script in a comment below. Solid work!

File: server/db/seed-postgres.js (hunk 3 above)
Comment (joshuabradley012) [reply server/db/seed-postgres.js:95]: Only about 10 minutes. I could make CSV files and use copy, but I like that I didn't

**Pull request merged**